test/message_bus/message_inbox/setup_message_inbox.js

```diff
@@ -0,0 +1,183 @@
+          setupParams.stateRoot,
+          setupParams.maxStorageRootItems,
+        ),
+        'Outbox address is 0.',
+      );
+    });
+
+    it('Should fail when state root provider is 0', async () => {
+      const stateRootProvider = '0x0000000000000000000000000000000000000000';
+
+      await Utils.expectRevert(
+        messageInbox.setupMessageInboxExternal(
+          setupParams.metachainId,
+          setupParams.messageOutbox,
+          setupParams.outboxStorageIndex,
+          stateRootProvider,
+          setupParams.maxStorageRootItems,
+        ),
+        'State root provider address is 0.',
+      );
+    });
+
+    it('Should fail when inbound channel identifier is 0', async () => {
+      await messageInbox.setInboundChannelIdentifier(
+        '0x0000000000000000000000000000000000000000000000000000000000000001',
+      );
+
+      await Utils.expectRevert(
+        messageInbox.setupMessageInboxExternal(
+          setupParams.metachainId,
+          setupParams.messageOutbox,
+          setupParams.outboxStorageIndex,
+          setupParams.stateRoot,
+          setupParams.maxStorageRootItems,
+        ),
+        'Message inbox is already setup.',
+      );
+    });
+  });
+
+  contract('Positive Tests', async () => {
+    it('Should successfully setup message inbox with required the parameters', async () => {
+      await messageInbox.setupMessageInboxExternal(
+        setupParams.metachainId,
+        setupParams.messageOutbox,
+        setupParams.outboxStorageIndex,
+        setupParams.stateRoot,
+        setupParams.maxStorageRootItems,
+      );
+
+      const messageOutbox = await messageInbox.messageOutbox.call();
+
+      assert.strictEqual(
+        messageOutbox,
+        setupParams.messageOutbox,
+        'Message Outbox from contract must be same as setupParams.messageOutbox',
+      );
+
+      const outboxStorageIndex = await messageInbox.outboxStorageIndex.call();
+
+      assert.strictEqual(
+        setupParams.outboxStorageIndex.eq(outboxStorageIndex),
+        true,
+        'Outbox Storage Index from contract must be same as Outbox Storage Index',
+      );
+
+      const MMB_CHANNEL_TYPEHASH = web3.utils.keccak256(
+        'MosaicMessageBusChannel(address outbox, address inbox)',
+      );
+
+      const MMB_DOMAIN_SEPARATOR_TYPEHASH = web3.utils.keccak256(
+        'MosaicMessageBus(string name,string version,bytes32 metachainId,bytes32 channelSeparator)',
+      );
+
+      const MMB_DOMAIN_SEPARATOR_NAME = 'Mosaic-Bus';
+      const MMB_DOMAIN_SEPARATOR_VERSION = '0';
+
+      const channelSeparator = web3.utils.keccak256(
+        web3.eth.abi.encodeParameters(
+          [
+            'bytes32',
+            'address',
+            'address',
+          ],
+          [
+            MMB_CHANNEL_TYPEHASH,
+            setupParams.messageOutbox,
+            messageInbox.address,
+          ],
+        ),
+      );
+
+      const expectedinboundChannelIdentifier = web3.utils.keccak256(
```

Comment:
This logic is already implemented in test/consensus-gateway/utils.js.
You can reuse it and avoid duplication.